pkg/sippyserver/metrics/metrics.go

```diff
@@ -132,6 +149,89 @@ func RefreshMetricsDB(dbc *db.DB, variantManager testidentification.VariantManag
 	return nil
 }
 
+func refreshComponentReadinessMetrics(client *bqclient.Client) error {
+	if client == nil || client.BQ == nil {
+		log.Infof("not generating component readiness metrics as we don't have a bigquery client")
+		return nil
+	}
+
+	// Sort our known GA releases, and get the most recent one: that's our base release
+	type releaseInfo struct {
+		Version *version.Version
+		Release string
+	}
+	var releases []releaseInfo
+	for release := range releaseloader.GADateMap {
+		v, err := version.NewVersion(release)
+		if err != nil {
+			log.WithError(err).Error("unparseable release " + release)
+			return err
+		}
+		releases = append(releases, releaseInfo{v, release})
+	}
+	sort.Slice(releases, func(i, j int) bool {
+		return releases[i].Version.LessThan(releases[j].Version)
+	})
+	mostRecentGA := releases[len(releases)-1].Release
+	log.Debugf("most recent GA is %q", mostRecentGA)
+	baseRelease := apitype.ComponentReportRequestReleaseOptions{
+		Release: mostRecentGA,
+		// Match what UI sends to API, although it's not correct TRT-1346
+		Start: releaseloader.GADateMap[mostRecentGA].AddDate(0, 0, -29),
+		End:   releaseloader.GADateMap[mostRecentGA].Add(-1 * time.Second),
+	}
+
+	// Get the next minor, that's our sample release
+	next, err := nextMinor(mostRecentGA)
+	if err != nil {
+		log.WithError(err).Error("couldn't calculate next minor")
+		return err
+	}
+	today := time.Now().UTC().Truncate(24 * time.Hour)
+	sampleRelease := apitype.ComponentReportRequestReleaseOptions{
+		Release: next,
+		Start:   today.AddDate(0, 0, -7),
+		// Match what UI sends to API, although it's not correct TRT-1346
+		End: today.Add(24 * time.Hour).Add(-1 * time.Second),
+	}
+
+	// Defaults - these are also hardcoded in the UI side
+	testIDOption := apitype.ComponentReportRequestTestIdentificationOptions{}
+	excludeOption := apitype.ComponentReportRequestExcludeOptions{
+		ExcludePlatforms: "openstack,alibaba,ibmcloud,libvirt,ovirt,unknown",
```

Comment:
It might be better to define those as const somewhere above with comments. It will be easier to find and modify.